pkg/apis/openebs.io/upgrade/v1alpha2/types.go

```diff
@@ -0,0 +1,169 @@
+/*
+Copyright 2019 The OpenEBS Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1alpha2
+
+import (
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// +genclient
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+// +resource:path=upgradetask
+// +k8s:openapi-gen=true
+
+// UpgradeTask represents an upgrade task
+type UpgradeTask struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty" protobuf:"bytes,1,opt,name=metadata"`
+	// Spec i.e. specifications of the upgradeTask
+	Spec UpgradeTaskSpec `json:"spec" protobuf:"bytes,2,opt,name=spec"`
+	// Status of upgradeTask
+	Status UpgradeTaskStatus `json:"status" protobuf:"bytes,3,opt,name=status"`
+}
+
+// UpgradeTaskSpec is the properties of an upgrade task
+type UpgradeTaskSpec struct {
+	FromVersion  string       `json:"fromVersion,omitempty" protobuf:"bytes,1,name=fromVersion"`
```

Comment:
Let us include other new generic parameters that we recently found like -- ImagePrefix, ImageTag

Reply:
Done